core/src/test/java/org/jclouds/rest/internal/RestAnnotationProcessorTest.java

```diff
@@ -1543,6 +1549,14 @@ public void testPathParamExtractor() throws Exception {
    }
 
    @Test
+   public void testPathParamEncoding() throws Exception {
+      Invokable<?, ?> method = method(TestPath.class, "onePathEncodedParam", String.class);
+      GeneratedHttpRequest request = processor.apply(Invocation.create(method,
+            ImmutableList.<Object> of("foo/bar")));
+      assertRequestLineEquals(request, "GET http://localhost:9999/foo%2Fbar HTTP/1.1");
```

Comment:
Use a more complex path in the test to also verify the fact that the entire path is not encoded. We need to verify that in the tests too.